src/transaction.ts

```diff
@@ -579,26 +717,39 @@ class Transaction extends DatastoreRequest {
   /**
    * This function parses results from a beginTransaction call
    *
-   * @param {RequestPromiseReturnType} response The response from a call to
-   * begin a transaction.
-   * @param {RunCallback} callback A callback that accepts an error and a
+   * @param {UserCallbackData<google.datastore.v1.IBeginTransactionResponse>} [response]
+   * The response data from a call to begin a transaction.
+   * @param {RunCallback} [callback] A callback that accepts an error and a
    * response as arguments.
    *
    **/
   #processBeginResults(
-    response: RequestPromiseReturnType,
+    response: UserCallbackData<google.datastore.v1.IBeginTransactionResponse>,
     callback: RunCallback
   ): void {
     const err = response.err;
     const resp = response.resp;
     if (err) {
       callback(err, null, resp);
     } else {
-      this.id = resp!.transaction;
+      this.#parseRunSuccess(response);
       callback(null, this, resp);
     }
   }
 
+  /**
+   * This function saves results from a successful beginTransaction call.
+   *
+   * @param {UserCallbackData<any>} [response] The response from a call to
+   * begin a transaction that completed successfully.
+   *
+   **/
+  #parseRunSuccess(response: UserCallbackData<any>) {
```

Comment:
We could change the response type here to `UserCallbackData<google.datastore.v1.IBeginTransactionResponse>`,  but then id would need to accept the `Buffer` type which would be a breaking change since the user can directly read `id`.